Updating SpecFlow to 2.1.0

lazytesting opened the pull request:
I updated SpecFlow and the other dependencies.

- updated SpecFlow from 1.9.0 to 2.1.0
- changed TestGenerator to work with the SpecFlow 2.1.0 
- updated Fluent.SeleniumWebdriver to 3.0
- updated to .Net 4.5 
- rewrote tests as the existing test where depending on external websites and failing

k0stya commented on FluentAutomation.Generator.SpecflowPlugin/FluentAutomationGenerator.cs:
```diff
@@ -15,7 +15,7 @@ private static void SaveTestClass(TestClassGenerationContext generationContext)
 			generationContext.ScenarioInitializeMethod.Statements.Add(
 				new CodeSnippetExpression("ScenarioContext.Current[ScenarioContext.Current.ScenarioInfo.Title] = this"));
-		}
+        }
```
Redundant formatting change

k0stya replied:
Use tabs instead of white-spaces to keep codebase consistent

k0stya commented on FluentAutomation.Generator.SpecflowPlugin/FluentAutomationTestGenerator.cs:
```diff
@@ -1,33 +1,82 @@
+                    }
+                default:
+                {
+                    throw new Exception(String.Format("Unknow unit test provider: {0}", unitTestProviderName));
```
Should be **NotImplementationException** instead of generic Exception

k0stya commented on FluentAutomation.Generator.SpecflowPlugin/Providers/NUnitTestGeneratorProvider.cs:
```diff
@@ -1,9 +1,11 @@
 {
-	public class NUnitTestGeneratorProvider : TechTalk.SpecFlow.Generator.UnitTestProvider.NUnitTestGeneratorProvider
+	public class 
+        NUnitTestGeneratorProvider : TechTalk.SpecFlow.Generator.UnitTestProvider.NUnitTestGeneratorProvider
```
Formatting

k0stya commented on NUnit.Selenium/PageObjectExamples/Pages/HomePage.cs:
```diff
@@ -0,0 +1,24 @@
+
+		public HomePage(FluentTest test) : base(test)
+		{
+            var currentPath = Path.GetDirectoryName(System.Reflection.Assembly.GetExecutingAssembly().GetName().CodeBase);
```
Formatting

k0stya commented:
Thanks for the contribution. Please fix minor issues with formatting, particularly use tabs instead of white-spaces and I'll merge the request.

lazytesting commented:
Thanks for your quick response! I'll fix those issues and let you know.

lazytesting commented:
Hi @k0stya I changed the exception and fixed the indenting for all the files which are not generated . Let me know if there is anything else I can do.
